### DISCUSSION TITLE
Helia with delegatedHTTPRouting and libp2p

### DISCUSSION THREAD

**Discussion by 2color** [ipfs/helia#660] [Q&A]
What is the expected behaviour of Helia with both `delegatedHTTPRouting` and `libp2pRouting` when libp2p is configured exclusively to use delegated routing? 

The use case for this is as follows:
1. You want both bitswap and http gateways block brokers
2. DHT client is disabled
3. You rely solely on a delegated routing endpoint

This came up as part of: https://github.com/ipfs-examples/helia-examples/pull/341

**Comment by achingbrain** [comment]
If you want bitswap you need libp2p and some sort of routing implementation.

If you've configured libp2p to use HTTP routing, and configured Helia to use libp2p, also configuring `delegatedHTTPRouting` at the Helia level seems redundant?
